server/nodejs/src/envConfig.ts

```diff
@@ -13,14 +22,33 @@ const envConfig = z.object({
   COUNSEL_API_HOST: z.string().default("https://sandbox-api.counselhealth.com"),
   JWT_SECRET: z.string().min(32),
   COUNSEL_WEBHOOK_SECRET: z.string(),
-  // This is the access code for the onboarding process which is a different flow from the main access code
-  ACCESS_CODE_COUNSEL_ONBOARDING: z.string().length(6),
-  // This is the API key for the onboarding process which is a different flow & therefore a different API key
-  COUNSEL_ONBOARDING_API_KEY: z.string(),
+  // JSON string mapping access codes to client/environment/API key configs
+  // Format: {"ACCESS_CODE": {"client": "main", "environment": "sandbox", "apiKey": "key123"}, ...}
+  ACCESS_CODE_CONFIGS: z
+    .string()
+    .transform((val) => {
+      try {
+        return JSON.parse(val);
+      } catch {
+        throw new Error("ACCESS_CODE_CONFIGS must be valid JSON");
+      }
+    })
+    .pipe(
+      z.record(z.string().length(6), AccessCodeConfigSchema).refine(
+        (configs) => Object.keys(configs).length > 0,
+        { message: "At least one access code configuration is required" }
+      )
+    ),
 });
 
 export const env = envConfig.parse(process.env);
 
 export function loadEnvConfig() {
   return envConfig.parse(process.env);
 }
+
+// Helper to get access code config
+export function getAccessCodeConfig(accessCode: string): z.infer<typeof AccessCodeConfigSchema> | null {
+  const config = env.ACCESS_CODE_CONFIGS[accessCode];
```

Comment:
make sure to transform the `accessCode` here into uppercase and trim

This is to ensure it works with `code23` and `CODE23`